### PR TITLE
implement health check in docker cluster, ticker based

### PR DESCRIPTION
**Pull request by runshenzhu**
Only the last commit is for health check.

The approach for health check is to add a go routine to poll container's health status periodically. If container becomes unhealthy, it will be shut down and the corresponding error will be thrown to upper layer.

### CODE REVIEW

**Comment by runshenzhu** [comment]
[this PR and comment](https://github.com/docker/swarmkit/pull/1007#issuecomment-228469570) is relevant.

File: daemon/cluster/executor/container/controller.go [hunk 1]
```diff
@@ -153,21 +154,69 @@ func (r *controller) Wait(pctx context.Context) error {
...
+		select {
+		case <-stop:
+		case errChan <- err:
 		}
-		if ec, ok := err.(exec.ExitCoder); ok {
-			ee.code = ec.ExitCode()
+	}()
+
+	go func() {
+		// polling health status every 3 seconds
```
Comment (dongluochen) [inline daemon/cluster/executor/container/controller.go]: Probing health status doesn't sound right to me. Engine should emit events on container healthiness. Cluster manager should consume the events.

**Comment by runshenzhu** [comment]
@dongluochen  Cluster doesn't implement the method that listens on, and consumes events from an event queue. 
Should I port [this method](https://github.com/docker/swarmkit/blob/master/agent/exec/container/adapter.go#L141) from swarmkit to docker engine in [here](https://github.com/docker/docker/blob/master/daemon/cluster/executor/container/adapter.go#L167)?

**Comment by dongluochen** [comment]
I think health check should be done with event stream. We should also add tests to make sure events are received and processed properly.

**Pull request closed without merging**